src/components/VoucherModal/index.tsx

```diff
@@ -65,7 +82,27 @@ const VoucherModal = ({ voucher, isOpen, onClose, isValidating }: Props) => {
 
           <hr />
 
-          <Text className={styles.footer}>Flash this eVoucher to redeem.</Text>
+          {voucher?.voucher_type !== "Dinamically allocated" && 
+            <Text className={styles.footer}>Flash this eVoucher to redeem.</Text>}
+
+          {voucher?.voucher_type === "Dinamically allocated" &&  !redeemed?.includes(voucher?.uuid || "") &&
```

Comment:
Do you think it would be better (for readability) to make `redeemed?.includes(voucher?.uuid || "")` as a Boolean variable i.e. `isRedeemed`, since it it also used down below. Similarly also for `voucher?.voucher_type === "Dinamically allocated"`. Considering that they are both used multiple times in this file.